policy-controller/src/admission.rs

```diff
@@ -57,23 +97,27 @@ pub fn routes(client: kube::Client) -> BoxedFilter<(impl warp::Reply,)> {
                 }
             };
             debug!(?rsp);
-            ok(reply::json(&rsp.into_review()))
+            json_response(rsp.into_review())
         })
-        .with(warp::trace::request())
-        .boxed()
+    }
 }
 
-type Review = kube::core::admission::AdmissionReview<DynamicObject>;
-type Request = kube::core::admission::AdmissionRequest<DynamicObject>;
-type Response = kube::core::admission::AdmissionResponse;
-
-#[inline]
-fn ok(reply: impl warp::Reply + 'static) -> Result<Box<dyn warp::Reply>, Infallible> {
-    Ok(Box::new(reply))
+fn json_response(rsp: AdmissionReview) -> Result<Response<Body>, BoxError> {
+    let bytes = serde_json::to_vec(&rsp)?;
+    Response::builder()
+        .status(http::StatusCode::OK)
+        .header(http::header::CONTENT_TYPE, "application/json")
+        .body(Body::from(bytes))
+        .map_err(|e| e.into())
 }
 
+type Review = kube::core::admission::AdmissionReview<DynamicObject>;
+type AdmissionRequest = kube::core::admission::AdmissionRequest<DynamicObject>;
+type AdmissionResponse = kube::core::admission::AdmissionResponse;
+type AdmissionReview = kube::core::admission::AdmissionReview<DynamicObject>;
+
 /// Parses a `Server` instance and it's namespace from the admission request.
```

Comment:
typo, not a blocker:
```suggestion
/// Parses a `Server` instance and its namespace from the admission request.
```